experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -257,3 +257,160 @@ function isomorphism(::Type{PermGroup}, W::WeylGroup; set_properties::Bool=true)
+Note that `map_word(::WeylGroupElem)` accepts the `genimgs_inv` keyword argument
+for consistency with other `map_word` methods, but ignores it because the
+generators of a Weyl group are always self-inverse.
+
+# Examples
+```jldoctest
+julia> W = weyl_group(:B, 3); imgs = [2, 3, 5];
+
+julia> map_word(one(W), imgs)
+1
+
+julia> map_word(W([1]), imgs)
+2
+
+julia> map_word(W([1]), imgs; init=7)
+14
+
+julia> map_word(W([1,2,1]), imgs)
+12
+
+julia> map_word(W([2,1,2]), imgs) # W([2,1,2]) == W([1,2,1])
+12
+
+julia> map_word(W([3, 2, 1, 3, 2, 3]), imgs)
+2250
+```
+"""
+function map_word(
+  w::WeylGroupElem, genimgs::Vector; genimgs_inv::Vector=genimgs, init=nothing
+)
+  @req length(genimgs) == number_of_generators(parent(w)) begin
+    "Length of vector of images does not equal rank of Weyl group"
+  end
+  return map_word(Int.(word(w)), genimgs; init=init)
+end
+
+@doc raw"""
+    parabolic_subgroup(W::WeylGroup, vec::Vector{<:Integer}, w::WeylGroupElem=one(W)) -> WeylGroup, Map{WeylGroup, WeylGroup}
+
+Return a Weyl group `P` and an embedding $f:P\to W$ such that $f(P)$ is
+the subgroup `U` of `W` generated by `[inv(w)*u*w for u in gens(W)[vec]]`.
+Further, `f` maps `gen(P, i)` to `inv(w)*gen(W, vec[i])*w`.
+The elements of `vec` must be pairwise distinct integers in
+`1:number_of_generators(W)` and `vec` must be non-empty.
+
+# Examples
+```jldoctest
+julia> W = weyl_group(:B, 3)
+Weyl group
+  of root system of rank 3
+    of type B3
+
+julia> P1, f1 = parabolic_subgroup(W, [1, 2])
+(Weyl group of root system of type A2, Map: P1 -> W)
+
+julia> f1(P1[1] * P1[2]) == W[1] * W[2]
+true
+
+julia> P2, f2 = parabolic_subgroup(W, [1, 2], W[1])
+(Weyl group of root system of type A2, Map: P2 -> W)
+
+julia> f2(P2[1]) == W[1] && f2(P2[2]) == W[1] * W[2] * W[1]
+true
+
+julia> P3, f3 = parabolic_subgroup(W, [1,3,2])
+(Weyl group of root system of type B3 (non-canonical ordering), Map: P3 -> W)
+
+julia> f3(P3[2]) == W[3]
+true
+```
+"""
+function parabolic_subgroup(W::WeylGroup, vec::Vector{<:Integer}, w::WeylGroupElem=one(W))
+  @req allunique(vec) "Elements of vector are not pairwise distinct"
+  @req all(i -> 1 <= i <= number_of_generators(W), vec) "Invalid indices"
+  cm = cartan_matrix(root_system(W))[vec, vec]
+  para = weyl_group(cm)
+  genimgs = [conj(W[i], w) for i in vec]
+  emb = function (u::WeylGroupElem)
+    return map_word(u, genimgs)
+  end
+  return para, MapFromFunc(para, W, emb)
+end
+
+@doc raw"""
+    parabolic_subgroup_with_projection(W::WeylGroup, vec::Vector{<:Integer}) -> WeylGroup, Map{WeylGroup, WeylGroup}, Map{WeylGroup, WeylGroup}
+
+Return a triple `(P, emb, proj)` that describes a factor of `W`, that is,
+a product of irreducible factors.
+Here `P, emb = `[`parabolic_subgroup`](@ref)`(W, vec)`
+and `proj` is the projection map from `W` onto `P`,
+which is a left-inverse of `emb`.
+If `check = true`, then it is checked whether `vec` actually describes
```

Comment:
```suggestion

If `check = true`, then it is checked whether `vec` actually describes
```
new paragraph for better readability.